### PR TITLE
add user manual for load-aware-scheduling and fine-grained-cpu-orches…

### PR DESCRIPTION
**Pull request by saintube**
Add user manuals for load-aware scheduling and fine-grained CPU orchestration.

Signed-off-by: saintube <saintube@foxmail.com>

### CODE REVIEW

File: docs/user-manuals/fine-grained-cpu-orchestration.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+1. Create an `nginx` deployment with the YAML file below. Set QoS classes of pods as [`LSE`](/docs/architecture/qos#definition) or [`LSR`](/docs/architecture/qos#definition).
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-lsr
+  labels:
+    app: nginx-lsr
+    koordinator.sh/qosClass: LSR
```
Comment (hormes) [inline docs/user-manuals/fine-grained-cpu-orchestration.md:104]: This label is useless and easy to misunderstand

File: docs/user-manuals/fine-grained-cpu-orchestration.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+2. Deploy the `nginx` deployment and check the scheduling result.
+
+```bash
+$ kubectl create -f nginx-deployment.yaml
+deployment/nginx-lsr created
+$ kubectl get pods -o wide | grep nginx
+nginx-lsr-59cf487d4b-jwwjv   1/1     Running   0       21s     172.20.101.35    node-0   <none>         <none>
+nginx-lsr-59cf487d4b-4l7r4   1/1     Running   0       21s     172.20.101.79    node-1   <none>         <none>
+nginx-lsr-59cf487d4b-nrb7f   1/1     Running   0       21s     172.20.106.119   node-2   <none>         <none>
+```
```
Comment (hormes) [inline docs/user-manuals/fine-grained-cpu-orchestration.md:138]: Add the strategy description of physical core spreading or binpack?

File: docs/user-manuals/load-aware-scheduling.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+```bash
+$ kubectl top node
+NAME     CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%
+node-0   6450m        19%    25091Mi         19%
+node-2   2687m        8%     17535Mi         13%
+node-1   3280m        10%    9631Mi          7%
+```
+
+In above order, `node-0` has the highest load, while `node-1` has the lowest load.
```
Comment (hormes) [inline docs/user-manuals/load-aware-scheduling.md:156]: node-2 is the lowest cpu... which is easy to misunderstand, set both cpu and memory are the lowest

File: docs/user-manuals/load-aware-scheduling.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+- Kubernetes >= 1.18
+- Koordinator >= 0.4
+
+### Installation
+
+Please make sure Koordinator components are correctly installed in your cluster. If not, please refer to [Installation](/docs/installation).
+
+### Configurations
+
+You can enable load-aware scheduling and configure the rules by modifying the ConfigMap `koord-scheduler-config` in the helm chart.
```
[the remark this answers is not in the document]
Comment (hormes) [reply docs/user-manuals/load-aware-scheduling.md:37]: For users who need deep insight, expand the detailed strategy

File: docs/user-manuals/fine-grained-cpu-orchestration.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx-lsr
+  template:
+    metadata:
+      name: nginx-lsr
+      labels:
+        app: nginx-lsr
+        koordinator.sh/qosClass: LSR # set the QoS class as LSR
```
Comment (eahydra) [inline docs/user-manuals/fine-grained-cpu-orchestration.md:115]: Annotation koordinator.sh/resource-spec needs to be setup up.

File: docs/user-manuals/fine-grained-cpu-orchestration.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+            numaAllocateStrategy: MostAllocated
+            # the scoring strategy
+            scoringStrategy:
+              # the kind of scoring strategy ('MostAllocated', 'BalancedAllocation', 'LeastAllocated')
+              type: MostAllocated
+              # the weights of each resource type
+              resources:
+              - name: cpu
+                weight: 2
+              - name: memory
```
Comment (eahydra) [inline docs/user-manuals/fine-grained-cpu-orchestration.md:66]: Memory also is not currently supported

File: docs/user-manuals/fine-grained-cpu-orchestration.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+          args:
+            apiVersion: kubescheduler.config.k8s.io/v1beta2
+            kind: NodeNUMAResource
+            # the CPU binding policy ('None', 'FullPCPUs', 'SpreadByPCPUs', 'ConstrainedBurst')
+            preferredCPUBindPolicy: FullPCPUs
+            # the NUMA allocation strategy ('MostAllocated', 'LeastAllocated', 'DistributeEvenly')
+            numaAllocateStrategy: MostAllocated
+            # the scoring strategy
+            scoringStrategy:
+              # the kind of scoring strategy ('MostAllocated', 'BalancedAllocation', 'LeastAllocated')
```
Comment (eahydra) [inline docs/user-manuals/fine-grained-cpu-orchestration.md:60]: BalancedAllocation is not currently supported.

File: docs/user-manuals/fine-grained-cpu-orchestration.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    profiles:
+      - schedulerName: koord-scheduler
+      - pluginConfig:
+        - name: NodeNUMAResource
+          args:
+            apiVersion: kubescheduler.config.k8s.io/v1beta2
+            kind: NodeNUMAResource
+            # the CPU binding policy ('None', 'FullPCPUs', 'SpreadByPCPUs', 'ConstrainedBurst')
+            preferredCPUBindPolicy: FullPCPUs
+            # the NUMA allocation strategy ('MostAllocated', 'LeastAllocated', 'DistributeEvenly')
```
Comment (eahydra) [inline docs/user-manuals/fine-grained-cpu-orchestration.md:56]: DistributeEvenly is not currently supported

File: docs/user-manuals/fine-grained-cpu-orchestration.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+2. Deploy the `nginx` deployment and check the scheduling result.
+
+```bash
+$ kubectl create -f nginx-deployment.yaml
+deployment/nginx-lsr created
+$ kubectl get pods -o wide | grep nginx
+nginx-lsr-59cf487d4b-jwwjv   1/1     Running   0       21s     172.20.101.35    node-0   <none>         <none>
+nginx-lsr-59cf487d4b-4l7r4   1/1     Running   0       21s     172.20.101.79    node-1   <none>         <none>
+nginx-lsr-59cf487d4b-nrb7f   1/1     Running   0       21s     172.20.106.119   node-2   <none>         <none>
```
Comment (eahydra) [inline docs/user-manuals/fine-grained-cpu-orchestration.md:137]: We can append a command to show how to view the CPUSet allocate results.

File: docs/user-manuals/fine-grained-cpu-orchestration.md (hunk 1 above)
Comment (saintube) [reply docs/user-manuals/fine-grained-cpu-orchestration.md]: removed

File: docs/user-manuals/fine-grained-cpu-orchestration.md (hunk 2 above)
Comment (saintube) [reply docs/user-manuals/fine-grained-cpu-orchestration.md:151]: added

File: docs/user-manuals/load-aware-scheduling.md (hunk 3 above)
Comment (saintube) [reply docs/user-manuals/load-aware-scheduling.md:161]: done

File: docs/user-manuals/load-aware-scheduling.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (saintube) [reply docs/user-manuals/load-aware-scheduling.md]: done

File: docs/user-manuals/fine-grained-cpu-orchestration.md [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+- Kubernetes >= 1.18
+- Koordinator >= 0.5
+
+### Installation
+
+Please make sure Koordinator components are correctly installed in your cluster. If not, please refer to [Installation](/docs/installation).
+
+### Configurations
+
+You can enable load-aware scheduling and configure the rules by modifying the ConfigMap `koord-scheduler-config` in the helm chart.
```
[the remark this answers is not in the document]
Comment (saintube) [reply docs/user-manuals/fine-grained-cpu-orchestration.md]: fixed

File: docs/user-manuals/fine-grained-cpu-orchestration.md (hunk 5 above)
Comment (saintube) [reply docs/user-manuals/fine-grained-cpu-orchestration.md:121]: fixed

File: docs/user-manuals/fine-grained-cpu-orchestration.md (hunk 7 above)
Comment (saintube) [reply docs/user-manuals/fine-grained-cpu-orchestration.md]: fixed

File: docs/user-manuals/fine-grained-cpu-orchestration.md (hunk 9 above)
Comment (saintube) [reply docs/user-manuals/fine-grained-cpu-orchestration.md:150]: done

File: docs/user-manuals/fine-grained-cpu-orchestration.md [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+      labels:
+        app: nginx-lsr
+        koordinator.sh/qosClass: LSR # set the QoS class as LSR
+      annotations:
+        # set the `resource-spec` annotation, configure the `preferredCPUBindPolicy`.
+        # `preferredCPUBindPolicy` indicates the CPU binding policy of the pod ('None', 'FullPCPUs', 'SpreadByPCPUs')
+        # - None: perform no exclusive policy
+        # - FullPCPUs: a bin-packing binding policy, prefer allocating full physical cores (SMT siblings)
+        # - SpreadByPCPUs: a spread binding policy, prefer allocating logical cores (SMT threads) evenly across physical cores (SMT siblings)
+        scheduling.koordinator.sh/resource-spec: '{"preferredCPUBindPolicy": "FullPCPUs"}'
```
Comment (hormes) [inline docs/user-manuals/fine-grained-cpu-orchestration.md:128]: 这样用户可能根本不会细看，而是直接复制粘贴。建议把策略分开，先讲 default 情况是怎样的，然后再给例子用户可以怎么设置策略，效果是怎样的。

Comment (saintube) [reply docs/user-manuals/fine-grained-cpu-orchestration.md:128]: 更新了，default 的逻辑先记下 todo

**Pull request merged**